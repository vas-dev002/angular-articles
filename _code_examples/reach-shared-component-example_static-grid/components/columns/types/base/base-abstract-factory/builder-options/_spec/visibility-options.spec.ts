import { IColumnMetaSG } from "../../../../../../../../../../../components/shared/aggrid/static-grid";
import { VisibilityOptions } from "..";

declare const expect: jest.Expect;

describe("[Option]: VisibilityOptions", () => {
	const meta = {
		field: "testField",
		cellClass: "testCellClass",
		headerName: "testHeaderName",
		hide: false,
		tooltipField: "testTooltipField",
		lockVisible: false,
		visible: true,
		cssClass: "testSlotClass"
	} as unknown as IColumnMetaSG;

	let visibilityOptions: VisibilityOptions;

	beforeEach(() => {
		visibilityOptions = new VisibilityOptions(meta);
	});

	describe("[VisibilityOptions instantiate]", () => {
		it("should be successfully created", () => {
			expect(visibilityOptions).toBeTruthy();
			expect(visibilityOptions).toMatchSnapshot();
		});
	});
});
