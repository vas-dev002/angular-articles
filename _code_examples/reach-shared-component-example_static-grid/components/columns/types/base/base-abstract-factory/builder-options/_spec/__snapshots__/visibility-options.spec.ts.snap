// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Option]: VisibilityOptions [VisibilityOptions instantiate] should be successfully created 1`] = `
VisibilityOptions {
  "hide": false,
  "lockVisible": false,
}
`;
